packages/storybook/src/stories/playground/customize-components.stories.tsx

```diff
@@ -465,3 +468,22 @@ CustomPanelInFooter.args = {
     Header: GridToolbar,
   },
 };
+
+export function CustomLayout() {
+  const apiRef = useApiRef();
+  apiRef.current.id = 123;
+
+  return (
+    <React.StrictMode>
+      <ApiRefProvider apiRef={apiRef}>
+        <div>
+          This button is outside Xgrid!
+          <FilterToolbarButton />
+        </div>
+        <div style={{ width: 300, height: 300 }}>
+          <XGrid rows={rows} columns={columns} apiRef={apiRef} />
```

Comment:
We shouldn't duplicate the source of truth, either it's the provider or a the grid

```suggestion
          <XGrid rows={rows} columns={columns} />
```